pkg/util/validation/limits.go

```diff
@@ -255,6 +258,8 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.AlertmanagerMaxDispatcherAggregationGroups, "alertmanager.max-dispatcher-aggregation-groups", 0, "Maximum number of aggregation groups in Alertmanager's dispatcher that a tenant can have. Each active aggregation group uses single goroutine. When the limit is reached, dispatcher will not dispatch alerts that belong to additional aggregation groups, but existing groups will keep working properly. 0 = no limit.")
 	f.IntVar(&l.AlertmanagerMaxAlertsCount, "alertmanager.max-alerts-count", 0, "Maximum number of alerts that a single tenant can have. Inserting more alerts will fail with a log message and metric increment. 0 = no limit.")
 	f.IntVar(&l.AlertmanagerMaxAlertsSizeBytes, "alertmanager.max-alerts-size-bytes", 0, "Maximum total size of alerts that a single tenant can have, alert size is the sum of the bytes of its labels, annotations and generatorURL. Inserting more alerts will fail with a log message and metric increment. 0 = no limit.")
+
+	f.Var(&l.EphemeralSeriesMatchers, "distributor.ephemeral-series-matchers", fmt.Sprintf("Lists of series matchers prefixed by the source. The source must be one of %s. If an incoming sample matches at least one of the matchers with its source it gets marked as ephemeral. The format of the value looks like: %s:{namespace=\"dev\"};%s:{host=\"server1\",namespace=\"prod\"}", strings.Join(ephemeral.ValidSourceStrings, ", "), ephemeral.SourceStringsBySource[ephemeral.API], ephemeral.SourceStringsBySource[ephemeral.RULE]))
```

Comment:
We should just use `ephemeral.API.String()` and `ephemeral.RULE.String()`. I assume reason why it was done this way was to avoid mixed receiver (non-pointer and pointer) on `ephemeral.Source` methods, which we can't avoid in that case. I'd rather ignore that warning and keep the code easier to read. Then we also don't need `ephemeral.SourceStringsBySource` map.

Reply:
> I assume reason why it was done this way was to avoid mixed receiver (non-pointer and pointer) on ephemeral.Source methods

correct.

>  I'd rather ignore that warning and keep the code easier to read. Then we also don't need ephemeral.SourceStringsBySource map.

ok, that's fine for me.

Reply:
this is done in 56a2f1ba8